src/Shared/BinarySerializer.cs

```diff
@@ -4,43 +4,79 @@
 //
 
 using System.IO;
-using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Web.SessionState;
+using ProtoBuf;
 
 namespace Microsoft.Web.Redis
 {
+
     public class BinarySerializer : ISerializer
     {
+        private readonly byte redisNull = 1;
+        private readonly byte initializeItem = 2;
+        private readonly byte str = 3;
         public byte[] Serialize(object data)
         {
-            if (data == null)
-            {
-                data = new RedisNull();
-            }
-            var binaryFormatter = new BinaryFormatter();
             using (var memoryStream = new MemoryStream())
```

Comment:
nit: no need to initialize this until after we've excluded the things that don't use the `MemoryStream`